### PR TITLE
Handle mocking of ModelType and ListType fields.

### PR DESCRIPTION
**Pull request by amibiz**

### CODE REVIEW

**Comment by amibiz** [comment]
For example,

```python
from schematics.models import Model
from schematics.types import StringType
from schematics.types.compound import ModelType


class Address(Model):
    country = StringType(required=True)
    city = StringType(required=True)


class Person(Model):
    name = StringType(required=True)
    address = ModelType(Address, required=True)


print Person.get_mock_object().to_primitive()
```

Will output the following,

`{'name': u'uyjZMFrXl', 'address': None}`

With the pull request, the *address* field gets mocked automatically,

`{'name': u'uyjZMFrXl', 'address': {'country': u'raBzFZLNtkNeiA', 'city': u'GMDI'}}`

This also applies to ListType.

**Comment by kracekumar** [comment]
Thanks, Will look into it and give further feedback.

File: schematics/types/compound.py [hunk 1]
```diff
@@ -143,6 +147,22 @@ def __init__(self, field, min_size=None, max_size=None, **kwargs):
     def model_class(self):
         return self.field.model_class
 
+    def _mock(self, context=None):
+        min_size = self.min_size or 1
+        max_size = self.max_size or 1
+
+        if min_size > max_size:
+            message = u'Minimum list size is greater than maximum list size.'
+            raise MockCreationError(message)
```
Comment (kracekumar) [inline schematics/types/compound.py]: `MockCreationError` isn't imported. Can you add some test case for this code ?

**Comment by bintoro** [comment]
Superseded by #306.

**Pull request closed without merging**